esp-hal/src/spi/master.rs

```diff
@@ -435,6 +440,35 @@ impl Address {
     }
 }
 
+/// SPI peripheral configuration
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct Config {
```

Comment:
I guess we can, I'll do that in one go in a follow-up. On the other hand, these will likely not change as much since existing peripherals won't gain new features.

Reply:
Whilst it's true it won't change in current chips, new chips may add new configuration. I agree though, lets do this in a follow up.

Reply:
Other chips' configuration won't affect the current ones, though. I mean, it wouldn't make sense introducing breaking changes just to add a field that won't be used :)

Reply:
I tend to agree since `#[non_exhaustive]` makes it impossible to use struct-update-syntax which in turn means we need that ugly way to create an instance like we currently have for `esp_hal::Config` (but yeah I understand _why_ we might want to have that)

Reply:
The point about new chips wasn't a great argument. If we can show that we have everything covered for every chip right now then I'm happy to relax this. Given that SPI is one of the core stable peripherals we plan on having for 1.0, I just wanted to hedge against breaking changes.

Reply:
I'm fine with `#[non_exhaustive]`. As `uart` already has a builder interface, we can use that here as well. As we had a few pieces on feedback for esp_hal::Config, we might as well implement a builder there. I imagine we can add more options to i2c for sure where the hardware has way more timeout features than we expose, I'm not sure what we'll end up doing with spi yet.